app/uk/gov/hmrc/selfservicetimetopay/controllers/ArrangementController.scala

```diff
@@ -44,103 +44,96 @@ class ArrangementController(ddConnector: DirectDebitConnector,
-      case EligibilityStatus(_, reasons) =>
-        Logger.info(s"Failed eligibility check because: $reasons")
-        Future.successful(Redirect(routes.SelfServiceTimeToPayController.getTtpCallUs()))
-    }
+  private def eligibilityCheck(taxpayer: Taxpayer)(implicit hc: HeaderCarrier) = {
+    eligibilityConnector.checkEligibility(EligibilityRequest(LocalDate.now(), taxpayer))
   }
 
-  def determineMisalignment: Action[AnyContent] = AuthorisedSaUser {
-    implicit authContext => implicit request =>
-      authorizedForSsttp {
+  def determineMisalignment: Action[AnyContent] = AuthorisedSaUser { implicit authContext =>
+    implicit request =>
+      authorizedForSsttp { submission =>
         val sa = authContext.principal.accounts.sa.get
 
         taxPayerConnector.getTaxPayer(sa.utr.utr).flatMap[Result] {
           _.fold(Future.successful(Redirect(routes.SelfServiceTimeToPayController.getTtpCallUs())))(t => {
-
-            sessionCache.get.flatMap[Result] {
-              _.fold(redirectToStart)(ttp => {
-                val newSubmission = ttp.copy(taxpayer = Some(t))
-                newSubmission match {
-                  case TTPSubmission(_, _, _, Some(Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(empty@Seq(), _, _, _, _, _), _) =>
-                    sessionCache.put(newSubmission.copy(calculatorData = CalculatorInput(startDate = LocalDate.now(),
-                      endDate = LocalDate.now().plusMonths(3).minusDays(1), debits = tpSA.debits))).map[Result] {
-                      _ => Redirect(routes.CalculatorController.getPaymentToday())
+            submission.fold(redirectToStart) { ttp =>
+              val newSubmission = ttp.copy(taxpayer = Some(t))
+              newSubmission match {
+                case TTPSubmission(_, _, _, Some(Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(empty@Seq(), _, _, _, _, _), _, _) =>
+                  sessionCache.put(newSubmission.copy(calculatorData = CalculatorInput(startDate = LocalDate.now(),
+                    endDate = LocalDate.now().plusMonths(3).minusDays(1), debits = tpSA.debits))).map {
+                    _ => Redirect(routes.CalculatorController.getPaymentToday())
+                  }
+                case TTPSubmission(None, _, _, Some(tp@Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(meDebits, _, _, _, _, _), _, _) =>
+                  sessionCache.put(newSubmission.copy(calculatorData = CalculatorInput(startDate = LocalDate.now(),
+                    endDate = LocalDate.now().plusMonths(3).minusDays(1), debits = tpSA.debits))).map {
+                    _ => Redirect(routes.CalculatorController.getPaymentToday())
+                  }
+                case TTPSubmission(_, _, _, Some(tp@Taxpayer(_, _, Some(tpSA))), _, _, CalculatorInput(meDebits, _, _, _, _, _), _, _)
+                  if areEqual(tpSA.debits, meDebits) =>
+                  eligibilityCheck(tp).flatMap { es =>
```

Comment:
PErform the eligibilityCheck and store the result off the back of this.
Slightly complicated by the need now to (potentially) change the scheduleDay.
A For comprehension may provide a better means of expressing this logic but this works for now.